python/pyspark/ml/regression.py

```diff
@@ -743,6 +743,18 @@ def treeWeights(self):
         """Return the weights for each tree"""
         return list(self._call_java("javaTreeWeights"))
 
```

Comment:
What is the advantage here of making tree ensemble models return regression trees? I understand that it is regression in 3 out of 4 cases, but it seems counterintuitive. Also, this property is left abstract in Scala and each of the four subclasses implements their own. I think @holdenk's suggestion is more general and more closely mirrors Scala.